### PR TITLE
feat: adding twist vehicle commands and states to proto API

### PR DESCRIPTION
**Pull request by strasdat**

### CODE REVIEW

File: protos/farm_ng/canbus/canbus.proto [hunk 1]
```diff
@@ -25,8 +25,17 @@ service CanbusService {
       returns (stream SendCanbusMessageReply) {}
   // Streams the states of all detected motors.
   // Motors not actively streaming their state will not appear in the StreamMotorStatesReply
+
   rpc streamMotorStates(StreamMotorStatesRequest)
       returns (stream StreamMotorStatesReply) {}
+
+  // Streams twist vehicle commands on the CAN bus
+  rpc twistVehicleCommand(TwistVehicleCommandRequest)
+    returns (TwistVehicleCommandReply) {}
```
Comment (Hackerman342) [inline protos/farm_ng/canbus/canbus.proto:34]: For sending CAN messages, the existing infrastructure is a [client streaming RPC](https://grpc.io/docs/what-is-grpc/core-concepts/#client-streaming-rpc). I'd tend towards doing it similarly as commands should be streamed at 20-50hz.  
```suggestion
  // Streams twist vehicle commands on the CAN bus
  rpc twistVehicleCommand(stream TwistVehicleCommandRequest)
    returns (TwistVehicleCommandReply) {}
```

File: protos/farm_ng/canbus/canbus.proto [hunk 2]
```diff
@@ -44,6 +53,32 @@ message SendCanbusMessageReply {
...
+
+message TwistVehicleStatesRequest {
+}
+
+
+message TwistVehicleStateReply {
+  double stamp = 1;
+  float linear_velocity_x_mps = 2;
+  float angular_velocity_rps = 3;
+}
```
Comment (Hackerman342) [inline protos/farm_ng/canbus/canbus.proto:79]: The only short coming of falling in line with the standard ROS message here is we lose the information of `[AmigaControlState](https://github.com/farm-ng/farm-ng-amiga/blob/main/py/farm_ng/canbus/packet.py#L30)`. Not sure this should stop us from going this direction, but we will want to get that information to ROS clients somehow.

Comment (Hackerman342) [reply protos/farm_ng/canbus/canbus.proto:79]: I guess that info could come from the command API if we keep the Unary RPC you prototyped above. If the robot is not in an auto control or auto ready state, the canbus service will not send the command and will respond with a False on the success bool.

Comment (Hackerman342) [reply protos/farm_ng/canbus/canbus.proto:79]: idk, I kind of think defining a ROS msg with the state is the way to go...

**Review by Hackerman342 — APPROVED** [review]

File: protos/farm_ng/canbus/canbus.proto (hunk 2 above)
Comment (Hackerman342) [reply protos/farm_ng/canbus/canbus.proto:79]: Something like:
```
[std_msgs/Header](http://docs.ros.org/en/noetic/api/std_msgs/html/msg/Header.html) header
[geometry_msgs/Twist](http://docs.ros.org/en/noetic/api/geometry_msgs/html/msg/Twist.html) twist

```

Comment (Hackerman342) [reply protos/farm_ng/canbus/canbus.proto:79]: Then users can grab the Twist from the message (trivial) and use it / send it along as needed

File: protos/farm_ng/canbus/canbus.proto (hunk 1 above)
Comment (edgarriba) [reply protos/farm_ng/canbus/canbus.proto:34]: following the existing canbus apis i'd name this method `sendTwistVehicleCommand`

File: protos/farm_ng/canbus/canbus.proto [hunk 3]
```diff
@@ -44,6 +53,32 @@ message SendCanbusMessageReply {
...
+
+message TwistVehicleCommandReply {
+  bool success = 1;
+}
+
+message TwistVehicleStatesRequest {
+}
+
+
+message TwistVehicleStateReply {
```
Comment (edgarriba) [inline protos/farm_ng/canbus/canbus.proto:75]: small thing to define, `State` or `States` ? just to be consistent with the api

File: protos/farm_ng/canbus/canbus.proto (hunk 2 above)
Comment (edgarriba) [reply protos/farm_ng/canbus/canbus.proto:79]: i vote also to compose the Request/Reply messages with more primitive messages

File: protos/farm_ng/canbus/canbus.proto (hunk 3 above)
Comment (strasdat) [reply protos/farm_ng/canbus/canbus.proto:75]: Well, it is streaming. The request is to stream "states". And it streams state by state one at a time.

**Pull request merged**